### PR TITLE
Replace HoS gun with a wrestlebelt on wrestlemap

### PR DESCRIPTION
**Pull request by JORJ949**
## About the PR 
On wrestlemap, the HoS' egun is instead a wrestling belt.

## Why's this needed? 
Funny feature for a funny map

### CODE REVIEW

File: code/obj/item/storage/backpack_belt_etc.dm [hunk 1]
```diff
@@ -1113,6 +1113,21 @@ TYPEINFO(/obj/item/storage/belt/wrestling)
...
+	contraband = 6
+
+	can_hold = list(/obj/item/baton, // Only include the things the HoS will be keeping on them.
+	/obj/item/gun/energy/taser_gun,
+	/obj/item/gun/energy/egun,
+	/obj/item/gun/energy/lawbringer,
+	/obj/item/gun/energy/wavegun,
+	/obj/item/gun/energy/tasersmg,
+	/obj/item/gun/energy/tasershotgun,
+	/obj/item/device/prisoner_scanner)
```
Comment (pgmzeta) [inline code/obj/item/storage/backpack_belt_etc.dm:1128]: less dupe pls (could still add tasershotgun in New like the shotgun belt does i guess)
```suggestion
	can_hold = /obj/item/storage/belt/security::can_hold
```

File: code/obj/storage/secure_closets.dm [hunk 2]
```diff
@@ -252,7 +252,11 @@ ADMIN_INTERACT_PROCS(/obj/storage/secure/closet, proc/break_open)
 	/obj/item/clothing/suit/armor/vest,
 	/obj/item/clothing/head/helmet/hardhat/security/hos,
 	/obj/item/clothing/glasses/sunglasses/sechud,
+#ifdef MAP_OVERRIDE_WRESTLEMAP
+	/obj/item/clothing/belt/wrestling/hos
```
Comment (pgmzeta) [inline code/obj/storage/secure_closets.dm:256]:
```suggestion
	/obj/item/storage/belt/wrestling/hos
```

**Review by pgmzeta — CHANGES_REQUESTED** [review]

File: code/obj/storage/secure_closets.dm [hunk 3]
```diff
@@ -252,7 +252,11 @@ ADMIN_INTERACT_PROCS(/obj/storage/secure/closet, proc/break_open)
 	/obj/item/clothing/suit/armor/vest,
 	/obj/item/clothing/head/helmet/hardhat/security/hos,
 	/obj/item/clothing/glasses/sunglasses/sechud,
+#ifdef MAP_OVERRIDE_WRESTLEMAP
+	/obj/item/storage/belt/wrestling/hos
```
Comment (pgmzeta) [inline code/obj/storage/secure_closets.dm:256]:
```suggestion
	/obj/item/storage/belt/wrestling/hos,
```

File: code/obj/item/storage/backpack_belt_etc.dm [hunk 4]
```diff
@@ -1113,6 +1113,18 @@ TYPEINFO(/obj/item/storage/belt/wrestling)
...
+	name = "Head of Security's championship belt"
+	desc = "The Head of Security's prized wrestling belt."
+	contraband = 6
+
+	New()
+		..()
+		var/list/holdable = /obj/item/storage/belt/security::can_hold
+		holdable += /obj/item/gun/energy/tasershotgun
+		src.can_hold = holdable
+		src.check_wclass = 1
```
Comment (pgmzeta) [inline code/obj/item/storage/backpack_belt_etc.dm:1126]: what you had here didn't work but this did; pls test changes

```suggestion
	can_hold = /obj/item/storage/belt/security::can_hold
	check_wclass = 1

	New()
		..()
		src.can_hold += /obj/item/gun/energy/tasershotgun
```

**Review by pgmzeta — APPROVED** [review]

**Comment by Retrino** [comment]
there's already one in the armory, dont really see the point of taking the hos's egun specifically

**Comment by JORJ949** [comment]
> there's already one in the armory, dont really see the point of taking the hos's egun specifically

The one in the armory is a fake one that only works in the ring and does no damage

**Comment by Retrino** [comment]
still dont understand why its going to the hos and not someone like the captain, the head authority on station. kind of just feeding into the main character energy stereotype that surrounds hos

**Comment by JORJ949** [comment]
> still dont understand why its going to the hos and not someone like the captain, the head authority on station. kind of just feeding into the main character energy stereotype that surrounds hos

I would be open to expanding it to replacing the captain's gun, but its easier to give a whitelist-only role a strong item than a job treated like AA staffie with a history of abuse (see old armory auth disk interaction)

**Comment by Chatauscours** [comment]
It's a gimmick map that's ran almost exclusively under admin supervision, i heavily doubt powergaming will be a problem if it's given to captains, considering the egun can be just as deadly than the belt if misused

**Comment by Tarmunora** [comment]
> still dont understand why its going to the hos and not someone like the captain, the head authority on station. kind of just feeding into the main character energy stereotype that surrounds hos

Agreed. This one is a swing and a miss I think.

**Pull request closed without merging**